ir/src/main/scala/org/scalajs/core/ir/InfoSerializers.scala

```diff
@@ -146,12 +146,15 @@ object InfoSerializers {
       val version = input.readUTF()
       val supported = ScalaJSVersions.binarySupported
       if (!supported.contains(version)) {
-        throw new IOException(
+        throw new IRVersionNotSupportedIOException(
             s"This version ($version) of Scala.js IR is not supported. " +
-            s"Supported versions are: ${supported.mkString(", ")}")
+            s"Supported versions are: ${supported.mkString(", ")}", version)
       }
 
       version
     }
   }
+
+  class IRVersionNotSupportedIOException(msg: String, val version: String)
```

Comment:
I don't think you need the `IO` in here. Subclasses of `IOException` usually don't contain `IO`.